package.json

```diff
@@ -34,7 +34,7 @@
     "prepublishOnly": "npm run build:package",
     "postpublish": "./docs_app/scripts/publish-docs.sh",
     "publish_docs": "./publish_docs.sh",
-    "test": "npm run compile && mocha --opts spec/support/default.opts \"dist/spec/**/*-spec.js\"",
+    "test": "npm run compile && cross-env TS_NODE_PROJECT=spec/support/tsconfig.json mocha --opts spec/support/default.opts \"dist/spec/**/*-spec.js\"",
```

Comment:
we removed dependency but specifying env var for ts-node?

Reply:
It's used to specify the aliases for the import paths `'rxjs/operators'`, etc. The `tsconfig-paths` `require` redirector needs it.

In the compile, the redirect need to be to the `src` directory, but here they need to be to the `dist/src` directory - because it's consuming JS.

We could use some other redirector - as that's all that's needed here - but I just left it using `tsconfig-paths`.

Reply:
Happy to use something else, if you can suggest another redirection package.

Reply:
Oh this reminds me reason I left ts-node, I used it for path mappings only. Need think little bit more - do we have some problem with current setup without this change?

Reply:
Yeah, without the change, I think there is a problem. Essentially, I think we need two sets of path mappings:

* When `tsc` builds the projects the path mappings must point it to the TypeScript files in `src` - the specs use import paths like `'rxjs/operators'`.
* When Mocha runs the tests, it's running the JavaScript files within `dist/spec`, so another set of path mappings are needed and they have to point to the files in `dist/src`.

Reply:
I removed `ts-node` because - with the change - it's not used. We only need the path mappings.

Reply:
> set of path mappings are needed and they have to point to the files

yes, this is something I am aware and agree eventually need to be revisited. my q was do we have some noticable / practical failure due to having this changes.

Reply:
There are no failures before the change and none after the change. The change doesn't fix anything that's broken. Instead, it makes sure that when the tests run, the run using the `tsc`-generated files that are in the `dist` directory.

Prior to the change, Mocha was running the JavaScript spec files in the `dist` directory, but was using `ts-node` to dynamically compile the source under test from the TypeScript files in the `src` directory. To me, this just seems so weird.

With the change, the tests and the source under test are loaded from the `dist` directory. AFAICT, this is faster - as you mentioned above "ts-node currently doesn't use incremental build meaningfully" - and seems much less strange.

I'm tempted to have a look for another path remapper, because the one that's used for the tests really has nothing to do with `ts-node` or with TypeScript.

Reply:
yeah, sounds like having better path remapping seems right way to go with cutting ts-node properly.